source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -0,0 +1,119 @@
+#include "extensions/filters/http/grpc_stats/grpc_stats_filter.h"
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+
+#include "common/grpc/codec.h"
+#include "common/grpc/common.h"
+#include "common/grpc/context_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcStats {
+
+namespace {
+
+class GrpcStatsFilter : public Http::StreamFilter {
```

Comment:
nit: I would recommend deriving from PassThroughFilter and then just overriding the methods you need?

Reply:
Done.